Source/MLX/Stream.swift

```diff
@@ -100,6 +100,11 @@ public final class Stream {
         mlx_free(ctx)
     }
 
+    /// Synchronize with the given stream
+    public func synchronize() {
+        mlx_synchronize(ctx)
+    }
+
```

Comment:
In python `synchronize` takes an optional stream so you can choose which stream to synchronize with: https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.synchronize.html#mlx.core.synchronize

Should we do the same / add an overload?

Reply:
You want it as a free function as well?  I can certainly add it.  This is a method on `Stream` so it doesn't need the argument.

Reply:
Oh I missed that it was a method. I think that's ok for now. We may want a free function which just synchronizes on the default stream (which is the most common use case), but we can also wait and see before adding it.